rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/Tile.java

```diff
@@ -0,0 +1,357 @@
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.rm.client.monitoring.views.compact;
+
+import static org.ow2.proactive_grid_cloud_portal.rm.client.NodeSource.*;
+import static org.ow2.proactive_grid_cloud_portal.rm.client.NodeSource.Host.*;
+
+import org.ow2.proactive_grid_cloud_portal.common.client.JSUtil;
+import org.ow2.proactive_grid_cloud_portal.rm.client.NodeSource;
+import org.ow2.proactive_grid_cloud_portal.rm.client.RMImages;
+
+import com.google.gwt.dom.client.Style;
+import com.google.gwt.event.dom.client.ContextMenuEvent;
+import com.google.gwt.user.client.DOM;
+import com.google.gwt.user.client.Event;
+import com.google.gwt.user.client.ui.Image;
+import com.smartgwt.client.widgets.Label;
+import com.smartgwt.client.widgets.layout.Layout;
+import com.smartgwt.client.widgets.menu.Menu;
+import com.smartgwt.client.widgets.menu.MenuItem;
+import com.smartgwt.client.widgets.menu.MenuItemSeparator;
+
+
+public class Tile extends Image {
+
+    private Host.Node node;
+
+    private Host host;
+
+    private NodeSource nodesource;
+
+    private Layout hover;
+
+    private Label hoverLabel;
+
+    private boolean dirty = true;
+
+    private CompactView compactView;
+
+    private CompactFlowPanel panel;
+
+    Tile(CompactView compactView, CompactFlowPanel panel, NodeSource ns) {
+        super(ns.getIcon());
+        this.nodesource = ns;
+        this.compactView = compactView;
+        this.panel = panel;
+        init();
+    }
+
+    Tile(CompactView compactView, CompactFlowPanel panel, Host host) {
+        super(host.isVirtual() ? RMImages.instance.host_virtual_16().getSafeUri().asString()
+                               : RMImages.instance.host_16().getSafeUri().asString());
+        this.host = host;
+        this.compactView = compactView;
+        this.panel = panel;
+        init();
+    }
+
+    Tile(CompactView compactView, CompactFlowPanel panel, Host.Node node) {
+        super(node.getIcon());
+        this.node = node;
+        this.compactView = compactView;
+        this.panel = panel;
+        init();
+    }
+
+    @Override
+    public void onBrowserEvent(Event event) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)